README.md

```diff
@@ -111,10 +111,8 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
     let publisher = service.publisher().create()?;
 
     while !SignalHandler::termination_requested() {
-        let mut sample = publisher.loan()?;
-        unsafe {
-            sample.as_mut_ptr().write(1234);
-        }
+        let sample = publisher.loan_uninit()?;
+        let sample = sample.write_payload(1234);
         publisher.send(sample)?;
```

Comment:
Why does this example work, maybe I am overlooking something, but it looks identical to the `compile_failure` example in `compiletests.rs`

Reply:
because it is using `write_payload` which consumes the sample and changes it from `MaybeUninit<T>` to `T`. The ` compiletests.rs` works on the `MaybeUninit<T>`. In that case `write` returns a `&T` but does not remove the  `MaybeUninit` from the sample.